app/controllers/attachment_references_controller.rb

```diff
@@ -0,0 +1,14 @@
+class AttachmentReferencesController < ApplicationController
+  load_resource :attachment_reference
+  before_action :authorize_resource
+
+  def show
+    redirect_to @attachment_reference.url
+  end
+
+  private
+
+  def authorize_resource
+    authorize!(:read, @attachment_reference.attachable)
```

Comment:
We authorised the attachable, like the forum post case, post is authorised and then we authorise the image in the post again ?